src/main/java/model/Cars.java

```diff
@@ -1,50 +1,59 @@
 package model;
 
-import dto.CarDto;
+import dto.RacingCarStateDto;
+import dto.RacingRoundStateDto;
 import dto.WinnerCarDto;
-import exception.DuplicateCarNameException;
-import java.util.Arrays;
+import exception.WrongRangeMovableNumberException;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
+import model.wrapper.Round;
 import utils.RacingNumberGenerator;
 
 public class Cars {
 
-    private static final String SEPARATOR = ",";
+    private static final int MIN_INCLUSIVE_VALUE = 0;
+    private static final int MAX_INCLUSIVE_VALUE = 9;
 
-    private final List<Car> cars;
+    private final Set<Car> cars;
 
-    public Cars(String inputNames) {
-        String[] carsName = inputNames.split(SEPARATOR);
-        List<Car> inputCars = Arrays.stream(carsName)
-                .distinct()
-                .map(Car::new)
-                .collect(Collectors.toList());
-
-        validateNameDuplication(carsName.length, inputCars.size());
-        cars = inputCars;
+    public Cars(Set<Car> cars) {
+        this.cars = cars;
     }
 
-    private void validateNameDuplication(int nameSize, int carSize) {
-        if (nameSize != carSize) {
-            throw new DuplicateCarNameException();
+    public List<RacingRoundStateDto> race(RacingNumberGenerator generator, Round round) {
+        List<RacingRoundStateDto> racingResult = new ArrayList<>();
+
+        while (round.isRacing()) {
+            racingResult.add(race(generator));
         }
+        return racingResult;
     }
 
-    public void race(RacingNumberGenerator generator) {
-        cars.forEach(car -> car.race(generator));
+    private RacingRoundStateDto race(RacingNumberGenerator generator) {
+        List<RacingCarStateDto> racingCarsStateDto = cars.stream()
+                .map(car -> car.race(generateValidMovableNumber(generator)))
```

Comment:
1단계에서는 Position, 2단계에서는 Cars 에 해당 genertor의 숫자 생성 역할을 두셨는데요!

무엇보다 '정답이 없는 문제'이고, 많이 고민해봤을 것 같아서 제 개인적인 생각을 남깁니다.
<**절대** 정답이 아님 주의>

저라면 다음처럼 해당 역할을 Car에 위임하여 처리할 것 같습니다.
```suggestion
                .map(car -> car.race(generator))
```

객체 지향은, 객체를 밖에서 조종한다기 보다는 각 객체에 책임을 주는 것이라 생각해요. 그래서 저는 Cars라는 관리 객체(?) 보다는 각 객체에 책임을 줄 것 같습니다. Position에게 넘겨 역할을 부여하는 것도 좋지만, 저라면 Position은 값객체 정도로 두고 (validation의 역할이나, 간단한 로직처리) 숫자 생성하여 입력하는 비즈니스 로직은 Car에 둘 것 같습니다.

Reply:
좋은 말씀인 것 같습니다!

Cars가 Car를 관리해야 하기 때문에 Cars에 의존성을 추가했는데, Generator가 결국 자동차를 전진시키는 역할을 담당하니 `일종의 엔진`으로 본다면 Car에 두는 게 더 자연스러울 것 같습니다.

어떤 객체가 어떤 책임을 가져야 할지 조금 더 고민해보겠습니다. 감사합니다!